src/logging.rs

```diff
@@ -87,9 +84,7 @@ pub fn setup_layers(
         .with_writer(std::io::stderr)
         .with_filter(EnvFilter::from_env("AZURE_INIT_LOG"));
 
-    let log_path = config
-        .map(|cfg| cfg.azure_init_log_path.path.clone())
-        .unwrap_or_else(|| PathBuf::from(DEFAULT_AZURE_INIT_LOG_PATH));
+    let log_path = config.azure_init_log_path.path.clone();
```

Comment:
Total nitpick, but since you're no longer using `DEFAULT_AZURE_INIT_LOG_PATH` as a fall back, you don't really need this clone since all the users of it take references; you can just replace `log_path` with `config.azure_init_log_path.path` references.